### ISSUE TITLE
android app crash

### ISSUE THREAD

**Issue by meiskam** [nextcloud/android#6957]
happens anytime I open the app now, had been working fine for about a year before this

recently (before this error) it's been claiming it can't access the server (from auto image upload, file name conflict), but it created about a dozen versions of that 1 file on the server, assuming once per time the notification showed up .. so obviously it has been accessing it

I haven't updated either server or client in a few months

************ CAUSE OF ERROR ************

java.lang.IllegalArgumentException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkParameterIsNotNull, parameter value
	at com.owncloud.android.datamodel.ArbitraryDataSet.<init>(Unknown Source:12)
	at com.owncloud.android.datamodel.ArbitraryDataProvider.getArbitraryDataSet(ArbitraryDataProvider.java:207)
	at com.owncloud.android.datamodel.ArbitraryDataProvider.storeOrUpdateKeyValue(ArbitraryDataProvider.java:67)
	at com.owncloud.android.operations.RefreshFolderOperation.updateDirectEditing(RefreshFolderOperation.java:313)
	at com.owncloud.android.operations.RefreshFolderOperation.updateCapabilities(RefreshFolderOperation.java:295)
	at com.owncloud.android.operations.RefreshFolderOperation.updateOCVersion(RefreshFolderOperation.java:270)
	at com.owncloud.android.operations.RefreshFolderOperation.run(RefreshFolderOperation.java:223)
	at com.owncloud.android.lib.common.operations.RemoteOperation.run(RemoteOperation.java:360)
	at java.lang.Thread.run(Thread.java:764)

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30130090
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: samsung
Device: heroqltespr
Model: SM-G930P
Id: R16NW
Product: heroqltespr

************ FIRMWARE ************
SDK: 26
Release: 8.0.0
Incremental: G930PVPSACSI2

(reported on my phone, will add more details later if requested)

**Comment by meiskam** [comment]
server:
```
Nextcloud
Version: 15.0.5.3
Apps installed: 37
Apps updates available: 8

PHP
Version: 7.3.6
Memory Limit: 512 MB
Max Execution Time: 3600
Upload max size: 10 GB

Database
Type: mysql
Version: 10.3.16
Size: 18.2 MB
```

## Server configuration detail

**Operating system:** Linux 4.19.107-Unraid #1 SMP Thu Mar 5 13:55:57 PST 2020 x86_64

**Webserver:** nginx/1.16.0 (fpm-fcgi)

**Database:** mysql 10.3.16

7.3.6
Modules loaded: Core, date, libxml, pcre, zlib, filter, hash, readline, Reflection, SPL, session, cgi-fcgi, bz2, ctype, curl, dom, fileinfo, ftp, gd, gmp, iconv, intl, json, ldap, mbstring, openssl, pcntl, PDO, pgsql, posix, standard, SimpleXML, smbclient, sodium, sqlite3, xml, xmlwriter, zip, exif, imap, mysqlnd, pdo_pgsql, pdo_sqlite, Phar, xmlreader, pdo_mysql, apcu, igbinary, redis, memcached, imagick, mcrypt, libsmbclient, Zend OPcache

**Nextcloud version:** 15.0.5 - 15.0.5.3

**Where did you install Nextcloud from:** unraid/docker - https://hub.docker.com/r/linuxserver/nextcloud/

<details><summary>Signing status</summary>

Array
(
)

</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - accessibility: 1.1.0
 - activity: 2.8.2
 - admin_audit: 1.5.0
 - bookmarks: 0.16.2
 - bruteforcesettings: 1.3.0
 - checksum: 0.4.2
 - cloud_federation_api: 0.1.0
 - comments: 1.5.0
 - dav: 1.8.1
 - federatedfilesharing: 1.5.0
 - federation: 1.5.0
 - files: 1.10.0
 - files_external: 1.6.0
 - files_pdfviewer: 1.4.0
 - files_rightclick: 0.11.1
 - files_sharing: 1.7.0
 - files_texteditor: 2.7.0
 - files_trashbin: 1.5.0
 - files_versions: 1.8.0
 - files_videoplayer: 1.4.0
 - gallery: 18.2.0
 - issuetemplate: 0.7.0
 - keeweb: 0.5.0
 - logreader: 2.0.0
 - lookup_server_connector: 1.3.0
 - metadata: 0.9.0
 - music: 0.9.3
 - notes: 2.5.1
 - notifications: 2.3.0
 - oauth2: 1.3.0
 - password_policy: 1.5.0
 - provisioning_api: 1.5.0
 - serverinfo: 1.5.0
 - sharebymail: 1.5.0
 - theming: 1.6.0
 - twofactor_backupcodes: 1.4.1
 - updatenotification: 1.5.0
 - workflowengine: 1.5.0
Disabled:
 - encryption
 - firstrunwizard
 - nextcloud_announcements
 - polls
 - richdocuments
 - support
 - survey_client
 - systemtags
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "memcache.local": "\\OC\\Memcache\\APCu",
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "***REMOVED SENSITIVE VALUE***",
        "***REMOVED SENSITIVE VALUE***",
        "***REMOVED SENSITIVE VALUE***",
        "***REMOVED SENSITIVE VALUE***"
    ],
    "dbtype": "mysql",
    "version": "15.0.5.3",
    "overwrite.cli.url": "https:\/\/***REMOVED SENSITIVE VALUE***",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "mysql.utf8mb4": true,
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "simpleSignUpLink.shown": false,
    "auth.bruteforce.protection.enabled": false,
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpmode": "smtp",
    "mail_sendmailmode": "pipe",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpsecure": "tls",
    "mail_smtpauth": 1,
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "587",
    "mail_smtpauthtype": "LOGIN",
    "maintenance": false,
    "theme": "",
    "loglevel": 2
}
```
</details>

**Are you using external storage, if yes which one:** yes, local

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

## Logs

<details><summary>Web server error log</summary>

```
nginx error.log is empty
```
</details>

<details><summary>Nextcloud log</summary>

```
most recent thing in nextcloud.log is from 2020-07-06 (2 months ago)

{
    "reqId": "iZLJuSE0C2R9B3NOYZjC",
    "level": 3,
    "time": "2020-07-06T14:30:16+00:00",
    "remoteAddr": "",
    "user": "--",
    "app": "PHP",
    "method": "",
    "url": "--",
    "message": "Doctrine\\DBAL\\DBALException: Failed to connect to the database: An exception occured in driver: SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo failed: Try again at \/config\/www\/nextcloud\/lib\/private\/DB\/Connection.php#64",
    "userAgent": "--",
    "version": "15.0.5.3"
}
```
</details>

**Comment by tobiasKaminsky** [comment]
This should be fixed with 3.13.1.
Please update and if it still occurs, please re-open.
